lib/bolt/config.rb

```diff
@@ -106,27 +113,31 @@ class Config
     DEFAULT_DEFAULT_CONCURRENCY = 100
 
     def self.default
-      new(Bolt::Project.new('.'), {})
+      new(Bolt::Project.create_project('.'), {})
     end
 
     def self.from_project(project, overrides = {})
-      data = {
-        filepath: project.config_file,
-        data: Bolt::Util.read_optional_yaml_hash(project.config_file, 'config')
-      }
+      conf = if project.project_file == project.config_file
+               project.data
+             else
+               Bolt::Util.read_optional_yaml_hash(project.config_file, 'config')
+             end
+
+      data = { filepath: project.config_file, data: conf }
 
       data = load_defaults(project).push(data).select { |config| config[:data]&.any? }
 
       new(project, data, overrides)
     end
 
     def self.from_file(configfile, overrides = {})
-      project = Bolt::Project.new(Pathname.new(configfile).expand_path.dirname)
-
-      data = {
-        filepath: project.config_file,
-        data: Bolt::Util.read_yaml_hash(configfile, 'config')
-      }
+      project = Bolt::Project.create_project(Pathname.new(configfile).expand_path.dirname)
```

Comment:
Hmm this is a little strange. If I run with `--configfile bolt.yaml` but there's a `bolt-project.yaml` next to it, it will use that instead?

Reply:
Yes, I said as much in the commit message:
> This means that users can specify --configfile path/to/bolt.yaml, and if there's a bolt-project.yaml with config
next to it it will not be loaded.

Reply:
I think the case where you want to do that is rare, and I think users who are using `bolt-project.yaml` and have config in it know what they're doing enough to work within that limitation. This is also pretty temporary (we'll hopefully have one config file soon), so I'm not Too Worried:tm:

Reply:
:+1: That's what I get for skimming the commit message. I think we can deprecate `--configfile` altogether before long.